src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -1501,6 +1501,81 @@ class LedgerRPC_test : public beast::unit_test::suite
         }
     }
 
+    void testTxDeliveredAmount()
+    {
+        testcase("Ledger Request, Tx Delivered Amount");
+
+        using namespace test::jtx;
+        using namespace std::chrono_literals;
+
+        Account const alice("alice");
+        Account const bob("bob");
+        Account const carol("carol");
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+
+        // number of expected 'delivered_amount' available
+        int numExpectedAvailable = 0;
+
+        Env env{*this};
+        env.fund(XRP(10000), alice, bob, carol, gw);
+        env.trust(USD(1000), alice, bob, carol);
+        env.close(NetClock::time_point{446000000s});
+
+        // normal payments
+        env(pay(gw, alice, USD(50)));
+        ++numExpectedAvailable;
+        env(pay(gw, alice, XRP(50)));
+        ++numExpectedAvailable;
+
+        // partial payment
+        env(pay(gw, bob, USD(9999999)), txflags(tfPartialPayment));
+        ++numExpectedAvailable;
+        env.require(balance(bob, USD(1000)));
+
+        // failed payment
+        env(pay(gw, carol, USD(9999999)), ter(tecPATH_PARTIAL));
+        // do not incExpected - the payment failed
+        env.require(balance(carol, USD(0)));
+
+        env.close();
+        std::string index;
+        Json::Value jvParams;
+        jvParams[jss::ledger_index] = 4u;
+        jvParams[jss::transactions] = true;
+        jvParams[jss::expand] = true;
+        auto const jtxn = env.rpc(
+            "json",
+            "ledger",
+            to_string(jvParams))[jss::result][jss::ledger][jss::transactions];
+        int numAvailable = 0;
+        for (auto const& t : jtxn)
+        {
+            bool const expectDeliveredAmount =
+                t["TransactionType"].asString() == "Payment" &&
+                t["metaData"]["TransactionResult"] == "tesSUCCESS";
+            bool const hasDeliveredAmount =
+                t["meta"].isMember("delivered_amount") &&
+                t["meta"]["delivered_amount"] != "unavailable";
+            BEAST_EXPECT(expectDeliveredAmount == hasDeliveredAmount);
+            if (hasDeliveredAmount)
+            {
+                ++numAvailable;
+                if (t["meta"].isMember("DeliveredAmount"))
+                {
+                    BEAST_EXPECT(
+                        t["meta"]["delivered_amount"] ==
+                        t["meta"]["DeliveredAmount"]);
+                }
+                else
+                {
+                    BEAST_EXPECT(t["Amount"] == t["meta"]["delivered_amount"]);
+                }
```

Comment:
Would there be any value in comparing the `tx` RPC result here? eg.
```Json::Value const meta = env.rpc("tx", t[jss::hash].asString())[jss::result][jss::meta];```

Reply:
Take a look at the new refactored tests in `Fixed in `Refactor tests and resolve PR feedback 3:`; I'm not sure if this resolves this issue or not (I do consider the tx result, but I'm not sure if that's what you meant or not). I'll leave this issue unresolved until I talk to you on Monday.

Reply:
I meant with every check iteration, cross-reference the relevant portion of an RPC "tx" (or "account_tx") result using the transaction hash from this "ledger" result. But since they both call insertDeliveredAmount, I don't believe there would be much benefit other than making sure the RPCs are reporting similar results. The refactor looks good as is, no change necessary.